src/PrestaShopBundle/Command/DebugCommand.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Command;
+
+use PrestaShop\PrestaShop\Adapter\Debug\DebugMode;
+use PrestaShop\PrestaShop\Adapter\LegacyContextLoader;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Configuration\Command\SwitchDebugModeCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * CLI command for getting and setting debug mode setting
+ */
+class DebugCommand extends Command
+{
+    public const STATUS_OK = 0;
+    public const STATUS_ERROR = 1;
+
+    /**
+     * @var SymfonyStyle
+     */
+    protected $io;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @var DebugMode
+     */
+    private $debugConfiguration;
+
+    public function __construct(CommandBusInterface $commandBus, DebugMode $debugConfiguration, LegacyContextLoader $legacyContextLoader)
+    {
+        parent::__construct();
+        $this->commandBus = $commandBus;
+        $this->debugConfiguration = $debugConfiguration;
+        $legacyContextLoader->loadGenericContext();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('prestashop:debug')
+            ->setDescription('Get or set debug mode')
+            ->addArgument('value', InputArgument::OPTIONAL, 'Value for debug mode, on/off, true/false, 1/0. If left out will just print the current state')
+        ;
+    }
+
+    protected function printDebugState()
```

Comment:
For visibility purposes, I'd use warning (or warning) for "ON" and success  for "OFF"

```php
// ex 1
$io->getErrorStyle()->warning('Debug mode is ON : don't forget to disable it ASAP');

//ex 2
$io->success('Debug mode is OFF. To enable it, execute the following :  "prestashop:debug on"
```

I believe it shouldn't be a success to be in debug mode in a production server

Reply:
@mickaelandrieu I disagree, the success is about the command succeeding "not about the debug state of the shop".

Just like the command `ls` on linux gives success return code even if there were no files listed. `ls` will only give error / warning status codes if it failed running (`pwd` does not exist or something similar)